arroyo/processing/strategies/run_task.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import time
+from collections import deque
+from concurrent.futures import Future, ThreadPoolExecutor
+from typing import Callable, Deque, Optional, Tuple, TypeVar
+
+from arroyo.processing.strategies.abstract import MessageRejected, ProcessingStrategy
+from arroyo.types import Commit, Message, TPayload
+
+logger = logging.getLogger(__name__)
+
+TOutput = TypeVar("TOutput")
+
+
+class RunTask(ProcessingStrategy[TPayload]):
```

Comment:
Nit: I feel the current naming is too generic and could cause problems. How about we named it `RunTaskInThreads` to make it more explicit in what it does.

My primary concern is that in the documentation it is mentioned that `This strategy can be used to run IO-bound tasks in parallel, then commit offsets.` Parallel could mean in threads or processes. The current implementation is using threads. That means people should ensure that the processing_function should avoid using objects in the processing_function which can be modified by different threads or protect it using locks. Might be worth adding these caveats in the docstring as well.

Reply:
Good point, updated